python/ray/serve/controller.py

```diff
@@ -519,6 +534,27 @@ async def __init__(self,
 
         asyncio.get_event_loop().create_task(self.run_control_loop())
 
+    async def wait_for_event(self, uuid: ResultId) -> FutureResult:
+        future = self.inflight_results[uuid]
+        result = await future
+        # NOTE(ilr) We have to delete this *after* we return to avoid a
+        # situation where the controller crashes *after* deleting and before
+        # returning to the API (and user).
+        asyncio.get_event_loop().call_soon(
+            lambda: self.inflight_results.pop(uuid))
+        return result
+
+    def _create_and_set_result(self, result: bool,
+                               goal_state: Dict[str, any]) -> ResultId:
+        fut = asyncio.get_event_loop().create_future()
+        fut.set_result(FutureResult(result, goal_state))
+        uuid = get_random_letters(length=20)
```

Comment:
nit: don't call these IDs UUIDs because they aren't :)

Reply:
Switched to UUID4s!